tools/kagami/src/genesis.rs

```diff
@@ -241,10 +259,248 @@ fn generate_synthetic(
                     ),
                 )
                 .into();
-                first_transaction.append_instruction(mint);
+                first_transaction.push_instruction(mint);
             }
         }
     }
 
     Ok(genesis)
 }
+
+/// Use `Kagami` to sign genesis block.
+#[derive(ClapArgs, Clone, Debug)]
+#[command(group = ArgGroup::new("private_key").required(true))]
+#[command(group = ArgGroup::new("public_key").required(true))]
+#[command(group = ArgGroup::new("format").required(false))]
+pub struct SignArgs {
+    /// The algorithm of the provided keypair
+    #[clap(default_value_t, long, short)]
+    algorithm: crypto::AlgorithmArg,
+    /// Private key (in string format) to sign genesis block
+    #[clap(long, group = "private_key")]
+    private_key_string: Option<String>,
+    /// Path to private key to sign genesis block
+    #[clap(long, group = "private_key")]
+    private_key_file: Option<PathBuf>,
+    /// Public key in multihash format of the corresponding private key
+    #[clap(long, group = "public_key")]
+    public_key_string: Option<String>,
```

Comment:
You can put a type which will deserialize to `PublicKey` from a string. (i.e. `PublicKey` itself!)